### PR TITLE
Make mini load/batch load, compaction, delete operation running.

### PR DESCRIPTION
**Pull request by chaoyli**
1. Mini load/batch load does not flush data to disk when finished.
   Now calling flush() when finishing writing data.
2. Base compaction/cumulative compaction should capture consistent rowsets
   before modify_rowsets.
3. Delete operation should modify tablet_meta when finished.
   Also, add this modification to memory data structure.

### CODE REVIEW

File: be/src/olap/delete_handler.cpp [hunk 1]
```diff
@@ -45,6 +45,34 @@ using google::protobuf::RepeatedPtrField;
...
+            LOG(WARNING) << "invalid condition. condition=" << ThriftDebugString(condition);
+            return OLAP_ERR_DELETE_INVALID_CONDITION;
+        }
+    }
+
+    // 存储删除条件
+    for (const TCondition& condition : conditions) {
+        string condition_str = construct_sub_predicates(condition);
+        del_pred->add_sub_predicates(condition_str);
+        LOG(INFO) << "store one sub-delete condition. condition=" << condition_str;
```
Comment (kangpinghuang) [inline be/src/olap/delete_handler.cpp]: is this log necessary? if true, use vlog

File: be/src/olap/push_handler.cpp [hunk 2]
```diff
@@ -191,7 +182,7 @@ OLAPStatus PushHandler::_do_streaming_ingestion(
                      << ", table=" << tablet->full_name()
                      << ", transaction_id=" << request.transaction_id;
         for (TabletVars& tablet_var : *tablet_vars) {
-            if (tablet_var.tablet.get() == NULL) {
+            if (tablet_var.tablet.get() == nullptr) {
```
Comment (kangpinghuang) [inline be/src/olap/push_handler.cpp]: tablet_var.tablet == nullptr

File: be/src/olap/tablet_meta.cpp [hunk 3]
```diff
@@ -371,12 +374,18 @@ OLAPStatus TabletMeta::delete_rs_meta_by_version(const Version& version) {
 
 OLAPStatus TabletMeta::modify_rs_metas(const vector<RowsetMetaSharedPtr>& to_add,
                                        const vector<RowsetMetaSharedPtr>& to_delete) {
-    for (auto del_rs : to_delete) {
+    for (auto rs_to_del : to_delete) {
         auto it = _rs_metas.begin();
         while (it != _rs_metas.end()) {
-            if (del_rs->version().first == (*it)->version().first
-                  && del_rs->version().second == (*it)->version().second) {
+            LOG(INFO) << "it version: " << (*it)->version().first << "-" << (*it)->version().second;
```
Comment (kangpinghuang) [inline be/src/olap/tablet_meta.cpp]: remove this log

**Review by kangpinghuang — CHANGES_REQUESTED** [review]

File: be/src/olap/tablet_meta.cpp (hunk 3 above)
Comment (chaoyli) [reply be/src/olap/tablet_meta.cpp]: OK

File: be/src/olap/push_handler.cpp (hunk 2 above)
Comment (chaoyli) [reply be/src/olap/push_handler.cpp]: OK

File: be/src/olap/delete_handler.cpp (hunk 1 above)
Comment (chaoyli) [reply be/src/olap/delete_handler.cpp]: This is used to know how many delete operations.
Take account of deleting operation is not so much, so LOG(INFO) is OK.

**Pull request merged**